homeassistant/components/esphome/__init__.py

```diff
@@ -83,6 +90,36 @@ def async_update_device_state(self, hass: HomeAssistantType) -> None:
         signal = DISPATCHER_ON_DEVICE_UPDATE.format(entry_id=self.entry_id)
         async_dispatcher_send(hass, signal)
 
+    async def async_load_from_store(self) -> List['EntityInfo']:
+        """Load the retained data from store and return de-serialized data."""
+        # pylint: disable= redefined-outer-name
+        from aioesphomeapi import COMPONENT_TYPE_TO_INFO, DeviceInfo
+
+        restored = await self.store.async_load()
+        if restored is None:
+            return []
+
+        device_info = DeviceInfo(**restored.pop('device_info'))
```

Comment:
Don't do this. Mention each key specifically. Otherwise people can't rollback if a key is added. (Lesson learned from auth)

Reply:
Good idea. I created a small helper method `_attr_obj_from_dict` which does that (I'm lazy and don't want to write a dict that stores all keys for each component type 😅)